packages/client-sdk-nodejs/src/internal/storage-data-client.ts

```diff
@@ -0,0 +1,337 @@
+    ];
+    return [
+      middlewaresInterceptor(
+        _loggerFactory,
+        middlewares,
+        middlewareRequestContext
+      ),
+      new HeaderInterceptorProvider(headers).createHeadersInterceptor(),
+      ClientTimeoutInterceptor(this.requestTimeoutMs),
+    ];
+  }
+
+  private createMetadata(storeName: string): Metadata {
+    const metadata = new Metadata();
+    metadata.set('store', storeName);
+    return metadata;
+  }
+
+  public async get(
+    storeName: string,
+    key: string
+  ): Promise<StorageGet.Response> {
+    try {
+      validateStoreName(storeName);
+    } catch (err) {
+      return this.cacheServiceErrorMapper.returnOrThrowError(
+        err as Error,
+        err => new StorageGet.Error(err)
+      );
+    }
+    this.logger.trace(
+      `Issuing 'get' request; store: ${storeName}, key: ${key}`
+    );
+    return await this.sendGet(storeName, key);
+  }
+
+  private async sendGet(
+    storeName: string,
+    key: string
+  ): Promise<StorageGet.Response> {
+    const request = new store._StoreGetRequest({
+      key: key,
+    });
+    const metadata = this.createMetadata(storeName);
+    return await new Promise((resolve, reject) => {
+      this.clientWrapper.getClient().Get(
+        request,
+        metadata,
+        {
+          interceptors: this.interceptors,
+        },
+        (err: ServiceError | null, resp) => {
+          const value = resp?.value?.value;
+          if (value) {
+            switch (value) {
+              case 'double_value': {
+                return resolve(
+                  new StorageGet.DoubleResponse(resp.value.double_value)
+                );
+              }
+              case 'string_value': {
+                return resolve(
+                  new StorageGet.StringResponse(resp.value.string_value)
+                );
+              }
+              case 'bytes_value': {
+                return resolve(
+                  new StorageGet.BytesResponse(resp.value.bytes_value)
+                );
+              }
+              case 'integer_value': {
+                return resolve(
+                  new StorageGet.IntegerResponse(resp.value.integer_value)
+                );
+              }
+              case 'none': {
+                return resolve(
+                  new StorageGet.Error(
+                    new UnknownError(
+                      'StorageGet responded with an unknown result'
+                    )
+                  )
+                );
+              }
+            }
+          } else {
+            this.cacheServiceErrorMapper.resolveOrRejectError({
+              err: err,
+              errorResponseFactoryFn: (e: SdkError) => new StorageGet.Error(e),
+              resolveFn: resolve,
+              rejectFn: reject,
+            });
+          }
+        }
+      );
+    });
+  }
+
+  public async put(
+    storeName: string,
+    key: string,
+    value: string | Uint8Array | number
```

Comment:
i don't think the number part of this is sufficient; if a user is writing data from this SDK and they intend for it to be all `double`s, and then someone is trying to read the data from another SDK they will potentially get some NOT_FOUND cases for any numbers that happened to be integers, right?

Reply:
cc: @pgautier404 because I think this is the same problem we were going to have in PHP with byte/string. Did you come up with anything you like there? I kind of think it's just going to have to be `putString`, `putDouble`, `putInt`, etc.?

cc: @malandis